hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/StandardSqmUpdateTranslator.java

```diff
@@ -263,13 +262,10 @@ public Expression resolveSqlExpression(
 
 					assert assignedPathJdbcCount == valueExprJdbcCount;
 
-					if ( valueExpression instanceof ColumnReference ) {
-						assert valueExprJdbcCount == 1;
-
-						assignments.add( new Assignment( (ColumnReference) valueExpression, valueExpression ) );
-					}
-					else {
-						throw new NotYetImplementedFor6Exception( "Support for composite valued assignments in an UPDATE query is not yet implemented" );
+					for (ColumnReference columnReference : targetColumnReferences) {
+						assignments.add(
+								new Assignment( columnReference, valueExpression )
+						);
```

Comment:
@sebersole I'm not sure that this is 100% correct, but it's certainly better than what was there before.

Reply:
Yes, generally implemented > !implemented ;)